From: Carlos Garnacho <carlosg@gnome.org>
Date: Fri, 10 Jul 2020 23:28:17 +0200
Subject: wayland: Replace cursor renderer usage with cursor tracker

We just want to know the pointer position, let's use something else
here.

Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1403>
---
 src/wayland/meta-wayland-dnd-surface.c | 12 +++++-------
 1 file changed, 5 insertions(+), 7 deletions(-)

diff --git a/src/wayland/meta-wayland-dnd-surface.c b/src/wayland/meta-wayland-dnd-surface.c
index 0786b99..5986680 100644
--- a/src/wayland/meta-wayland-dnd-surface.c
+++ b/src/wayland/meta-wayland-dnd-surface.c
@@ -73,16 +73,14 @@ static MetaLogicalMonitor *
 dnd_surface_find_logical_monitor (MetaWaylandActorSurface *actor_surface)
 {
   MetaBackend *backend = meta_get_backend ();
-  MetaCursorRenderer *cursor_renderer =
-    meta_backend_get_cursor_renderer (backend);
+  MetaCursorTracker *cursor_tracker =
+    meta_backend_get_cursor_tracker (backend);
   MetaMonitorManager *monitor_manager =
      meta_backend_get_monitor_manager (backend);
-  graphene_point_t pointer_pos;
+  int x, y;
 
-  pointer_pos = meta_cursor_renderer_get_position (cursor_renderer);
-  return meta_monitor_manager_get_logical_monitor_at (monitor_manager,
-                                                      pointer_pos.x,
-                                                      pointer_pos.y);
+  meta_cursor_tracker_get_pointer (cursor_tracker, &x, &y, NULL);
+  return meta_monitor_manager_get_logical_monitor_at (monitor_manager, x, y);
 }
 
 static double
